File: chrome/browser/tab/java/src/org/chromium/chrome/browser/tab/TabId.java
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.tab;

import static java.lang.annotation.ElementType.TYPE_USE;
import static java.lang.annotation.RetentionPolicy.SOURCE;

import org.chromium.build.annotations.NullMarked;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

/** Annotation to mark a tab id. */
@Retention(SOURCE)
@Target(TYPE_USE)
@NullMarked
public @interface TabId {}

File: chrome/browser/ui/tabs/tab_strip_api/tab_id.cc
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/ui/tabs/tab_strip_api/tab_id.h"

namespace tabs_api {

bool operator==(const TabId& a, const TabId& b) {
  return a.Type() == b.Type() && a.Id() == b.Id();
}

}  // namespace tabs_api
